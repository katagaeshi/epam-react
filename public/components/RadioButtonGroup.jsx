import React from 'react';
import PropTypes from 'prop-types';
import '../styles/RadioButtonGroup.css';

const getCheckedProperty = (checked, option) => (checked && option === checked);

class RadioButtonGroup extends React.Component {
  constructor(props) {
    super(props);
    this.handleOptionChange = this.handleOptionChange.bind(this);
    this.onUpdate = this.props.onUpdate;
    this.state = {
      selectedOption: this.props.checked,
    };
  }

  handleOptionChange(changeEvent) {
    const selectedOption = changeEvent.target.value;
    this.setState({
      selectedOption,
    });
    this.onUpdate(selectedOption);
  }

  render() {
    const options = this.props.options.map((option, idx) => {
      const id = this.props.idPrefix + idx;
      const key = id;
      return (
        <div key={key} className="RadioButton">
          <input
            type="radio"
            value={option}
            id={id}
            checked={getCheckedProperty(this.state.selectedOption, option)}
            onChange={this.handleOptionChange}
          />
          <label htmlFor={id}>{option}</label>
        </div>
      );
    });

    return options;
  }
}

RadioButtonGroup.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  idPrefix: PropTypes.string.isRequired,
  checked: PropTypes.string,
  onUpdate: PropTypes.func,
};

RadioButtonGroup.defaultProps = {
  checked: null,
  onUpdate: () => {},
};

export default RadioButtonGroup;
