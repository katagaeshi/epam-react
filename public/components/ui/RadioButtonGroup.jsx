// @flow

import React from 'react';
import injectSheet from 'react-jss';

const styles = {
  radioButton: {
    display: 'inline-block',
    height: '20px',
  },

  label: {
    color: 'white',
    'padding-right': '10px',
    'padding-left': '10px',
    'padding-top': '2px',
    'padding-bottom': '1px',
  },

  input: {
    visibility: 'hidden',
    '&:not(:checked) + label': {
      'background-color': 'black',
    },
    '&:checked + label': {
      'background-color': 'pink',
    },
  },
};

const getCheckedProperty = (checked, option) => (checked && option === checked);

type Props = {
  options: Array<string>,
  idPrefix: string,
  checked: string,
  onUpdate: () => {},
  classes: {
    radioButton: string,
    input: string,
    label: string,
  },
};

const RadioButtonGroup = (props: Props) => {
  const options = props.options.map((option, idx) => {
    const id = props.idPrefix + idx;
    const key = id;
    const {
      radioButton,
      input,
      label,
    } = props.classes;
    return (
      <div key={key} className={radioButton}>
        <input
          className={input}
          type="radio"
          value={option}
          id={id}
          checked={getCheckedProperty(props.checked, option)}
          onChange={props.onUpdate}
        />
        <label className={label} htmlFor={id}>{option}</label>
      </div>
    );
  });
  return options;
};

RadioButtonGroup.defaultProps = {
  checked: null,
};

export default injectSheet(styles)(RadioButtonGroup);
