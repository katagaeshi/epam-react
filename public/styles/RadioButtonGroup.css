.RadioButton {
  display: inline-block;
  height: 20px;
}

.RadioButton > input {
  visibility: hidden;
}

.RadioButton > label {
  color: white;
  padding-right: 10px;
  padding-left: 10px;
  padding-top: 2px;
  padding-bottom: 1px;
}

.RadioButton > input:not(:checked) + label {
  background-color: black;
}

.RadioButton > input:checked + label {
  background-color: pink;
}
