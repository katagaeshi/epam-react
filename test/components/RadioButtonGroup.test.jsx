import React from 'react';
import renderer from 'react-test-renderer';
import RadioButtonGroup from './../../public/components/RadioButtonGroup';

describe('RadioButtonGroup', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(
        <RadioButtonGroup
          options={['Pikachu', 'Charmander']}
          checked="Pikachu"
          idPrefix="radioButtonTest"
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
