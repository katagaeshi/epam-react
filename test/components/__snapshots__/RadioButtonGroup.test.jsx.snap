// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup Snapshot 1`] = `
Array [
  <div
    className="RadioButton"
  >
    <input
      checked={true}
      id="radioButtonTest0"
      onChange={[Function]}
      type="radio"
      value="Pikachu"
    />
    <label
      htmlFor="radioButtonTest0"
    >
      Pikachu
    </label>
  </div>,
  <div
    className="RadioButton"
  >
    <input
      checked={false}
      id="radioButtonTest1"
      onChange={[Function]}
      type="radio"
      value="Charmander"
    />
    <label
      htmlFor="radioButtonTest1"
    >
      Charmander
    </label>
  </div>,
]
`;
